Framework/Algorithms/test/ScaleXTest.h

```diff
@@ -110,20 +114,17 @@ class ScaleXTest : public CxxTest::TestSuite {
 
       for (size_t j = 0; j < xsize; ++j) {
         if (factor > 0) {
```

Comment:
While we're looking at efficiency, should the `if` statement be moved out of the innermost loop? If one takes i=0 and i=1 out of the loop, can the branch is completely removed.

I'm open to having this done as a separate issue/PR.